Preserve folder structure of Photos.app

alexantao opened the pull request:
This is a huge change.
the original program extracts all albums from Photos.app and creates all of them in the ROOT of OUTPUT dir. If there are any album with the same name they will be merged. 
In this version, the folder structure that was created in Photos.app are now replicated as folders under output_dir. The albums inside the folders are also replicated and the photos are copied inside the albums they belongs. No merge anymore, if a photo is present on several albums it will be replcated to the right album inside a folder, even if there's another album with the same name on another folder.
Photos not present on any album are places on root of output dir. 

For that, several changes were made in four files:
- two new JSON files are created (albums.json and folders.json) to help decide where to send each photo. (**folder_structure.py** and **albums_data.py**)
- **extract_photos.py** now stores albums IDs, instead of albums names.
- those albums IDs are translated with those JSON files and the folder/album are composed to copy the file. (**album_folder.py**) 

I didin't think yet about code optimization. It's a step ahead. 
Altrhough my editor has PEP8 checking integrated, I ran autoppep8 using format.sh
Tested on a small sample database of my own.

alexantao committed:
Merge pull request #1 from orangeturtle739/master

Merge from origin

alexantao committed:
Generates a JSON file containing the Folder Structure os Photos.app
Ignores some toplevels folders. Only user folders are stores.
to be used to recreate the folder structure with albuns inside.

alexantao committed:
Merge pull request #2 from orangeturtle739/master

Atualizando meu fork com o original.

alexantao committed:
Generates a JSON file containing the Folder Structure os Photos.app
Ignores some toplevels folders. Only user folders are stores.
to be used to recreate the folder structure with albuns inside.

alexantao committed:
Generates a JSON file containing the information about Albums
on Photos.app
Ignores albums inside specified toplevels folders.
Only user folders albums are stored.

alexantao committed:
Merge branch 'master' into albums_inside_folders

alexantao committed:
Finish folders copy operations.
Now the folders are replicated as they are on Photos.app
Albums are created inside their folders.
Photos copied inside their albums (if present in multiple albums,
they are replicated for each of them)
Photos without albums are stored on root output dir.

alexantao committed:
removal of prints

alexantao committed:
removal of prints

alexantao committed:
autopep8 ran

alexantao commented:
Oh, just remembered. Didi not removed the insertion of Album TAG on album. I think set_exif does this, will take a look. 
also, I did not updated the photos_extract.py, but the right sequence should be (must create another argument, that's  the final destination of the files) : 

`
def run(lib_dir, temp_dir, final_dir):
     output('Extracting photos')
     extract_photos.run(lib_dir, temp_dir)
     output('Cleaning album names') 
     clean_albums.run(temp_dir)
     output('Setting EXIF metadata')
     set_exif.run(temp_dir)
     output('Moving albums to folders...')
     folder_structure.run(lib_dir, temp_dir)
     album_data.run(lib_dir, temp_dir)
     album_folder.run(temp_dir, output_dir)

`

alexantao committed:
Normalize Path to all OS

orangeturtle739 commented:
This looks cool! I'll look when I have a chance.

orangeturtle739 commented:
This looks good! Can you please update `photos_extract.py`, and then I'll merge. Also, is there anything in the `README` that should be updated after this change?

alexantao committed:
My contribution on README and photos_export.py general Script.
Now it does all work, asking the DIGIKAM user if want to run
group_versions script.

First extract all photos to a subdirectory "temporarydirectory" of
output_dir to process it later. After the last script, the temporary
directory will keep all JSON files only.

alexantao commented:
Updated README and photos_extract.py script.

orangeturtle739 commented:
Awesome!
